commandapi-platforms/commandapi-bukkit/commandapi-bukkit-core/src/main/java/dev/jorel/commandapi/arguments/MapArgument.java

```diff
@@ -238,268 +214,293 @@ public CommandAPIArgumentType getArgumentType() {
 		return CommandAPIArgumentType.MAP;
 	}
 
-	@SuppressWarnings("unchecked")
 	@Override
 	public <Source> LinkedHashMap<K, V> parseArgument(CommandContext<Source> cmdCtx, String key, CommandArguments previousArgs) throws CommandSyntaxException {
-		String rawValues = cmdCtx.getArgument(key, String.class);
+		StringReader reader = new StringReader(cmdCtx.getArgument(key, String.class));
 		LinkedHashMap<K, V> results = new LinkedHashMap<>();
+		if (reader.getRemainingLength() == 0) return results;
+
+		K builtKey = null;
+
+		// Read through the keys and values
+		Set<String> givenKeys = new HashSet<>();
+		Set<String> givenValues = new HashSet<>();
+		List<String> unusedKeys = new ArrayList<>(keyList.results);
+		List<String> unusedValues = new ArrayList<>(valueList.results);
+
+		boolean isKey = true;
+		while (true) {
```

Comment:
Hm, it probably says `while(true)` because that's what I first put when planning the method, when I wasn't sure what the exit condition would be. Switching to `while(reader.canRead())` makes sense and should be easy given that immediately after, `if (reader.getRemainingLength() == 0)`, the method gets broken out of with `return startSuggestions(...`.